File: 프로그래머스/level2/62048.py
def euclidean(a, b):
    if b == 0:
        return a
    big = max(a, b)
    small = min(a, b)
    a, b = divmod(big, small)
    return 0 + euclidean(small, b)
    
def solution(w,h):
    total = w * h
    d = euclidean(w, h)
    erased_square = (w // d) + (h // d) - 1
    return total - erased_square * d

print(solution(5, 5))

File: 프로그래머스/level2/62048.js
const divmod = (a, b) => {
  return [parseInt(a/b), a % b]
}

const euclidean = (a, b) => {
  if (b === 0) {
    return a
  }
  let big = Math.max(a, b)
  let small = Math.min(a, b)
  
  let [c, d] = divmod(big, small)
  return 0 + euclidean(small, d)
}

const solution = (w, h) => {
  const total = w * h
  const d = euclidean(w, h)
  return total - (parseInt(w/d) + parseInt(h/d) - 1) * d
}
console.log(solution(8, 12))